specifyweb/frontend/js_src/lib/components/Statistics/StatsSpec.tsx

```diff
@@ -0,0 +1,502 @@
+              ],
+            },
+          },
+          speciesCount: {
+            label: statsText('species'),
+            spec: {
+              type: 'Querybuildstat',
+              tableName: 'Taxon',
+              fields: [
+                {
+                  path: formattedEntry,
+                },
+                {
+                  path: 'rankId',
+                  operStart: queryFieldFilters.equal.id,
+                  startValue: '220',
+                  isDisplay: false,
+                },
+              ],
+            },
+          },
+        };
+      },
+    },
+    localityGeography: {
+      label: statsText('localityGeography'),
+      categories(backendStatsResult) {
+        return {
+          localityCount: {
+            label: statsText('localities'),
+            spec: {
+              type: 'Querybuildstat',
+              tableName: 'Locality',
+              fields: [
+                {
+                  path: formattedEntry,
+                },
+                {
+                  path: 'localityId',
+                  operStart: queryFieldFilters.any.id,
+                },
+              ],
+            },
+          },
+          geographyEntryCount: {
+            label: statsText('geographyEntries'),
+            spec: {
+              type: 'Querybuildstat',
+              tableName: 'Geography',
+              fields: [
+                {
+                  path: formattedEntry,
+                },
+                {
+                  path: 'geographyId',
+                  operStart: queryFieldFilters.any.id,
+                },
+              ],
+            },
+          },
+          countriesCount: {
+            label: statsText('countries'),
+            spec: {
+              type: 'Backendstat',
+              value: backendStatsResult?.countries,
+            },
+          },
+          georeferencedLocalityCount: {
+            label: statsText('georeferencedLocalities'),
+            spec: {
+              type: 'Querybuildstat',
+              tableName: 'Locality',
+              fields: [
+                {
+                  path: formattedEntry,
+                },
+                {
+                  path: 'latitude1',
+                  operStart: queryFieldFilters.empty.id,
+                  isNot: true,
+                  isDisplay: false,
+                },
+              ],
+            },
+          },
+        };
+      },
+    },
+    typeSpecimens: {
+      label: statsText('typeSpecimens'),
+      categories: (backendStatsResult) =>
+        modifyBackendResult<'typeSpecimens'>(backendStatsResult, formatNumber),
+    },
+    catalogStats: {
+      label: statsText('computerization'),
+      categories() {
+        const local = new Date();
+        const initialDate = getDateInputValue(local);
+        local.setDate(local.getDate() - 7);
+        const initialDateSevenDaysPast = getDateInputValue(local);
+        local.setDate(local.getDate() - 23);
```

Comment:
This is bug prone (not intuitive what 23 is), and assumes every month is 30 days long.

Instead, do something like this:
```
const weekAgo = new Date();
weekAgo.setDate(weekAgo.getDate() - 7);

const monthAgo = new Date();
monthAgo.setDate(monthAgo.getMonth() - 1);

const yearAgo = new Date();
yearAgo.setDate(yearAgo.getFullYear() - 1);
```